### PR TITLE
[SYTO-666] Clicking Tooltip Button Opens Global Nav

### PR DESCRIPTION
**Pull request by hkekeli**
https://jira.nationalgeographic.com/browse/SYTO-666

-git checkout branch
- gulp serve
- Visit http://localhost:3000/components

- Click on the tooltip button to open a modal

### CODE REVIEW

File: app/modules/tooltips/_styles.scss [hunk 1]
```diff
@@ -90,3 +90,18 @@ $mt2_arrowheight: .625rem;
     }
   }
 }
+
+
+// Tooltip Modal
+
+.mt2_tooltip-modal-btn {
```
Comment (xiwcx) [inline app/modules/tooltips/_styles.scss]: `mt2_tooltip-modalbtn`, `mt2_tooltip-modalcontent`...

**Comment by krlospelaez** [comment]
@hkekeli @xiwcx I think we should reuse existing tooltip instead of create a new one for modals. Pass another pestle option to determinate when to use the bubble tooltip or a modal.

```html
<div data-pestle-module="Tooltip">
    <script type="text/json" data-pestle-options>
    {
        "tooltipContent" : "Hello World, inside a modal!",
        "modal": true
    }
    </script>
</div>
```

**Comment by xiwcx** [comment]
@krlospelaez i back that, @hkekeli make sense? seem doable?

**Comment by hkekeli** [comment]
@krlospelaez @xiwcx Done!

File: app/modules/tooltips/scripts/modal.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+import React, { Component, PropTypes }  from 'react';
+import * as ReactDOM from 'react-dom';
```
Comment (alexrqs) [inline app/modules/tooltips/scripts/modal.jsx]: don't see where are you using ReactDOM

File: app/modules/tooltips/scripts/modal.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import classNames from 'classnames';
```
Comment (alexrqs) [inline app/modules/tooltips/scripts/modal.jsx]: don't see where are you using this

File: app/modules/tooltips/scripts/modal.jsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+export class SimpleModal extends Component {
```
Comment (alexrqs) [inline app/modules/tooltips/scripts/modal.jsx]: This component don't have any methods, sounds like the perfect case for pure/stateless react components

File: app/modules/tooltips/scripts/MTTooltip.jsx [hunk 5]
```diff
@@ -3,8 +3,41 @@
...
+}
+
+class TooltipBubble extends Component{
+  static propTypes = {
+    tooltipContent: PropTypes.string,
+    placement: PropTypes.oneOf(['top', 'left', 'right', 'bottom'])
+  };
+
+  static defaultProps = {
+    placement: "top"
```
Comment (alexrqs) [inline app/modules/tooltips/scripts/MTTooltip.jsx]: lint for double quotes is not working, ;)

File: app/modules/tooltips/scripts/MTTooltip.jsx [hunk 6]
```diff
@@ -69,9 +98,51 @@ class Tooltip extends Component {
...
+    };
+
+    this.open = this.open.bind(this);
+    this.close = this.close.bind(this);
+  }
+  open(){
+    this.setState({active: true});
+  }
+  close(){
+    this.setState({ active: false});
```
Comment (alexrqs) [inline app/modules/tooltips/scripts/MTTooltip.jsx]: As the modal never will be open and closed at the same time, this scenario have a better formula using a toggleVisibility method that only apply 
```js 
this.setState({active: !this.state.active})
```

File: app/modules/tooltips/scripts/MTTooltip.jsx [hunk 7]
```diff
@@ -69,9 +98,51 @@ class Tooltip extends Component {
...
+  }
+  renderModal(){
+    const tooltipContent = this.props.tooltipContent ? this.props.tooltipContent : '';
```
Comment (alexrqs) [inline app/modules/tooltips/scripts/MTTooltip.jsx]: dont worry about spending memory on ternary the prop is a string so you can just `this.props.tooltipContent || ''`

File: app/modules/tooltips/scripts/MTTooltip.jsx (hunk 5 above)
Comment (hkekeli) [inline app/modules/tooltips/scripts/MTTooltip.jsx]: you mean it should be `placement: 'top'`?

File: app/modules/tooltips/scripts/MTTooltip.jsx (hunk 7 above)
Comment (hkekeli) [inline app/modules/tooltips/scripts/MTTooltip.jsx]: perfect

File: app/modules/tooltips/scripts/MTTooltip.jsx (hunk 5 above)
Comment (alexrqs) [inline app/modules/tooltips/scripts/MTTooltip.jsx]: right :)

**Comment by hkekeli** [comment]
@asabaylus Why did you change the title on this? Have the requirements changed?

**Comment by pmourelle** [comment]
LGTM!

**Pull request merged**